init.lua

```diff
@@ -169,61 +169,63 @@ function ranks.update_nametag(player)
 			prefix = ""
 		end
 
-		player:set_nametag_attributes({
-			text = prefix..name,
-		})
+		if player then
+			player:set_nametag_attributes({
+				text = prefix..name,
+			})
+		end
 
 		return true
 	end
 end
 
--- [function] Set player rank
-function ranks.set_rank(player, rank)
-	if type(player) == "string" then
-		player = minetest.get_player_by_name(player)
+-- [function] Set player rank TODO
+function ranks.set_rank(name, rank)
+	if type(name) == "string" then
+		player = minetest.get_player_by_name(name)
```

Comment:
Again, this statement should be effectively flipped. See other comments to the same end.
Also, I see `TODO` in the comment above. Is something else still to be done or should that be removed?

Reply:
Done!
I've removed the TODO too; although I cannot remember why I put it there xD
It's been a long time since I've looked at the code.

But I tested it anyway; just to make sure it is working correctly.